finance_dl/ultipro_google.py

```diff
@@ -171,23 +181,24 @@ def get_statement_table():
     def get_existing_statements(self):
         existing_statements = set()
         if os.path.exists(self.output_directory):
-            for name in os.listdir(self.output_directory):
-                m = re.fullmatch(
-                    r'([0-9]{4})-([0-9]{2})-([0-9]{2})\.statement-([0-9A-Z]+)\.pdf',
-                    name)
-                if m is not None:
-                    date = datetime.date(
-                        year=int(m.group(1)),
-                        month=int(m.group(2)),
-                        day=int(m.group(3)))
-                    statement_number = m.group(4)
-                    existing_statements.add((date, statement_number))
-                    logger.info('Found existing statement %r %r', date,
-                                statement_number)
-                else:
-                    logger.warning(
-                        'Ignoring extraneous file in existing statement directory: %r',
-                        os.path.join(self.output_directory, name))
+            for subdir, dirs, files in os.walk(self.output_directory):
+                for name in files:
+                    m = re.fullmatch(
```

Comment:
Again, seems partly replaceable with `glob.glob(os.path.join(output_directory, '**statement*.pdf'))`.
The double-asterisk should crawl all subfolders iirc.
Just a suggestion.

Reply:
In this case I do think switching to `glob` improves readability (and reduces nesting by making the if-exists check redundant). Thanks!